src/Database/Query.php

```diff
@@ -2013,7 +2014,8 @@ public function getIterator(): Traversable
             $results = $this->all();
         }
 
-        /** @var array $results */
+        $results = is_array($results) ? $results : iterator_to_array($results);
```

Comment:
The local variable assignments are for psalm which doesn't realize that $this->_results will always be set by this point since all() is called.

Reply:
Why use `iterator_to_array()` instead of directly returning the results? Only when it's an array it needs to be turned into a `Traversable`.